Service/ZGWService.php

```diff
@@ -405,12 +435,73 @@ public function downloadInhoudHandler(array $data, array $configuration): array
     }//end downloadInhoudHandler()
 
 
+
+    /**
+     * Handles gebruiksrecht POST logic.
+     *
+     * @param array $data          The data passed by the action.
+     * @param array $configuration The configuration of the action.
+     * 
+     * @return array Http response.
+     */
+    public function gebruiksrechtHandler(array $data, array $configuration): array
+    {
+        $this->data = $data;
+
+        // If not a POST this action has nothing to do.
+        if ($data['method'] !== 'POST') {
+            return $this->data;
+        }
+
+        $gebruiksrecht = $data['body'];
+
+        // Validate informatieobject
+        $explodedInfoObject = explode('/', $gebruiksrecht['informatieobject']);
+        if (
+            isset($gebruiksrecht['informatieobject']) == false || is_string($gebruiksrecht['informatieobject']) === false || 
+            (Uuid::isValid($gebruiksrecht['informatieobject']) === false && Uuid::isValid(end($explodedInfoObject)) === false)
+           ) {
```

Comment:
_(Just my preference, you can ignore this one if you want)_

This if statement does not look very easy to read. I would do something like this:
```
if (isset($gebruiksrecht['informatieobject']) == false
    || is_string($gebruiksrecht['informatieobject']) === false
    || (
        Uuid::isValid($gebruiksrecht['informatieobject']) === false
        && Uuid::isValid(end($explodedInfoObject)) === false
    )
)
```